src/text-field.jsx

```diff
@@ -331,7 +331,8 @@ const TextField = React.createClass({
       inputProps.onChange = this._handleInputChange;
     }
     if (this.props.children) {
-      inputElement = React.cloneElement(this.props.children, {...inputProps, ...this.props.children.props});
+      let childInputStyle = this.mergeStyles(inputStyle, this.props.children.style);
```

Comment:
Well, I think that #1896 will be reviewed and merged before this PR. But as you want